packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -393,19 +479,22 @@ export class AskarWallet implements Wallet {
    * @returns A signature for the data
    */
   public async sign({ data, key }: WalletSignOptions): Promise<Buffer> {
+    let keyEntry: KeyEntryObject | null | undefined
     try {
       if (keyTypeSupportedByAskar(key.keyType)) {
         if (!TypedArrayEncoder.isTypedArray(data)) {
           throw new WalletError(`Currently not supporting signing of multiple messages`)
         }
-        const keyEntry = await this.session.fetchKey({ name: key.publicKeyBase58 })
+        keyEntry = await this.session.fetchKey({ name: key.publicKeyBase58 })
 
         if (!keyEntry) {
           throw new WalletError('Key entry not found')
         }
 
         const signed = keyEntry.key.signMessage({ message: data as Buffer })
 
+        keyEntry.key.handle.free()
```

Comment:
:( we should be careful where we'll allow this free handling. Ideally we split this from the normal AFJ code, or we e.g. look at something like sesssions for a key and will close all keys in a session at once. I just don't think handle logic should be exposed from the askar library

Reply:
Not for now ofc.

Reply:
Yeah I really don't like having anything related to handles in AFJ or any other consumer of askar/anoncreds/indy-vdr, except maybe for more 'global' objects like sessions or connections. We already are having some fun discussions about this with @blu3beri and the _opinionating_ thing 😛